src/Databases/AlterDatabaseCommands.cpp

```diff
@@ -0,0 +1,413 @@
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ASTCreateQuery.h>
+#include <Parsers/ASTColumnDeclaration.h>
+#include <Parsers/ASTConstraintDeclaration.h>
+#include <Parsers/ASTIndexDeclaration.h>
+#include <Parsers/ASTTableOverrides.h>
+#include <Parsers/formatAST.h>
+#include <Common/logger_useful.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+    extern const int INVALID_ALTER_DATABASE_QUERY;
+    extern const int TABLE_OVERRIDE_ALREADY_EXISTS;
+    extern const int TABLE_OVERRIDE_NOT_FOUND;
+}
+
+void DB::AlterDatabaseSettingCommand::prepare(const DatabasePtr &, ContextPtr, AlterDatabaseApplyData & apply_data)
+{
+    for (const auto & change : settings_changes)
+    {
+        // Allow duplicate settings as long as the values are the same.
+        if (!apply_data.merged_settings_changes.insertSetting(change.name, change.value))
+        {
+            if (Field * found = apply_data.merged_settings_changes.tryGet(change.name); found && *found != change.value)
+            {
+                throw Exception(
+                    ErrorCodes::INVALID_ALTER_DATABASE_QUERY,
+                    "ALTER DATABASE MODIFY SETTING: multiple conflicting values for setting '{}'",
+                    change.name);
+            }
+        }
+    }
+    for (const auto & reset : settings_resets)
+    {
+        // Do not allow mixing MODIFY and RESET for the same setting
+        Field found_value;
+        if (!apply_data.merged_settings_changes.tryGet(reset, found_value))
+            apply_data.merged_settings_changes.push_back({reset, Field()});
+        else if (found_value.getType() != Field::Types::Null)
+            throw Exception(
+                ErrorCodes::INVALID_ALTER_DATABASE_QUERY,
+                "ALTER DATABASE RESET SETTING: can not mix MODIFY and RESET for setting '{}'",
+                reset);
+    }
+}
+
+void AlterDatabaseSettingCommand::apply(DatabasePtr &, ContextPtr, AlterDatabaseApplyData &)
+{
+    /// Nothing to do here, settings are batch-applied in AlterDatabaseSettings::applyAll()
+}
+
+AlterDatabaseTableOverrideCommand::AlterDatabaseTableOverrideCommand(ASTAlterCommand::Type type_, const ASTPtr & override_ast_)
+    : override_ast(override_ast_), type(type_), table_override(override_ast->as<const ASTTableOverride>())
+{
+}
+
+namespace
+{
+    template <typename T>
+    std::vector<const T *> to_const_elements(std::vector<T *> input)
+    {
+        std::vector<const T *> output(input.begin(), input.end());
+        return output;
+    }
+
+    ASTTableOverride * tryGetExistingOverride(const ASTCreateQuery * create, const String & override_name)
+    {
+        const ASTTableOverrideList * existing_table_overrides = create ? create->table_overrides : nullptr;
+        if (auto tmp = existing_table_overrides ? existing_table_overrides->tryGetTableOverride(override_name) : nullptr)
+            return tmp->as<ASTTableOverride>();
```

Comment:
A bit overcomplication to my taste, isn't it simpler:
```suggestion
        if (!create || !create->table_overrides)
            return nullptr;
        if (auto table_override = create->table_overrides->tryGetTableOverride(override_name))
            return table_override->as<ASTTableOverride>();
```